bsmetadata/preprocessing_tools/website_desc_utils.py

```diff
@@ -0,0 +1,40 @@
+from collections import defaultdict
+from typing import Optional
+
+import nltk
+from wikipedia2vec.dump_db import DumpDB
+
+
+class WebsiteDescUtils:
+    def __init__(self, path_wiki_db) -> None:
+        self.cache = defaultdict(str)
+        self.wiki_dump_db = DumpDB(path_wiki_db)
+        self.redirects_map = {
+            key.lower(): value for key, value in self.wiki_dump_db.redirects()
+        }  # loading all redirect information: takes ~10s
+
+    def fetch_wikipedia_title_from_keyword(self, keyword: str) -> str:
+        title = self.redirects_map.get(
+            keyword, keyword.split(".")[0].capitalize()
+        )  # fallback to default for cases where domain is not recognized. We'll try to hit the db with the exact keyword directly (e.g. rightmove.com -> Rightmove) Capitalizing since wikipedia titles are so
+        return title
+
+    def fetch_wikipedia_description_for_title(self, title: str) -> Optional:
+        try:
+            text = self.wiki_dump_db.get_paragraphs(title)[0].text
+            text = nltk.sent_tokenize(text)[0]  # Picking the first sentence
```

Comment:
Would it maybe make sense to remove information in brackets? For example, the first sentence in the Wikipedia article on Wikipedia itself is `Wikipedia (/ˌwɪkɪˈpiːdiə/ (listen) wik-ih-PEE-dee-ə or /ˌwɪki-/ (listen) wik-ee-) is a free content, multilingual online encyclopedia written and maintained by a community of volunteers through a model of open collaboration, using a wiki-based editing system.` At least in this case, I would think that the part in brackets (`/ˌwɪkɪˈpiːdiə/ (listen) wik-ih-PEE-dee-ə or /ˌwɪki-/ (listen) wik-ee-`) is completely useless for the model but will probably require many tokens. Any thoughts on that?

Reply:
Sounds like a great idea!

If it saves you some time @shanyas10 , here is a little code snippet that will remove the text in parentheses:
```python
import re

text = re.sub(r"\((?:[^)(]|\([^)(]*\))*\)", "", text)
```

On @timoschick's example, it will output: `Wikipedia  is a free content, multilingual online encyclopedia written and maintained by a community of volunteers through a model of open collaboration, using a wiki-based editing system.`

Reply:
Makes a lot of sense @timoschick . 
Thank you for the snippet @SaulLu . I've made the changes accordingly